server/handler/base.go

```diff
@@ -146,32 +146,43 @@ func (b *Base) Evaluate(ctx context.Context, installationID int64, trigger commo
 
 	fetchedConfig, err := b.ConfigFetcher.ConfigForPR(ctx, prctx, client)
 	if err != nil {
-		return errors.WithMessage(err, fmt.Sprintf("failed to fetch policy: %s", fetchedConfig))
+		return errors.WithMessage(err, fmt.Sprintf("Failed to fetch configuration: %s", fetchedConfig))
 	}
 
-	return b.EvaluateFetchedConfig(ctx, prctx, client, fetchedConfig, trigger)
+	evaluator, err := b.ValidateFetchedConfig(ctx, prctx, client, fetchedConfig, trigger)
+
+	if evaluator == nil {
+		return err
+	}
+
+	_, err = b.EvaluateFetchedConfig(ctx, prctx, client, evaluator, fetchedConfig)
+	return err
 }
 
-func (b *Base) EvaluateFetchedConfig(ctx context.Context, prctx pull.Context, client *github.Client, fetchedConfig FetchedConfig, trigger common.Trigger) error {
+func (b *Base) ValidateFetchedConfig(ctx context.Context, prctx pull.Context, client *github.Client, fetchedConfig FetchedConfig, trigger common.Trigger) (common.Evaluator, error) {
 	logger := zerolog.Ctx(ctx)
 
 	if fetchedConfig.Missing() {
-		logger.Debug().Msgf("policy does not exist: %s", fetchedConfig)
-		return nil
+		logger.Debug().Msg(fetchedConfig.Description())
+
+		return nil, errors.New(fetchedConfig.Description())
 	}
 
 	if fetchedConfig.Invalid() {
-		logger.Warn().Err(fetchedConfig.Error).Msgf("invalid policy: %s", fetchedConfig)
-		err := b.PostStatus(ctx, prctx, client, "error", fetchedConfig.Description())
-		return err
+		logger.Warn().Err(fetchedConfig.Error).Msg(fetchedConfig.Description())
+		b.PostStatus(ctx, prctx, client, "error", fetchedConfig.Description())
```

Comment:
You mentioned that we're no longer handling the error from `PostStatus`. I think we do need to do something here, but I agree that not returning the error makes it easier to return the other, more meaningful errors. 

Maybe we can change `PostStatus` to not return `error` and to always log the error instead? In general, there's not much we can do if we fail to post a status and there shouldn't be much harm in finishing out any remaining work, in the few cases where posting the status isn't the last step.

Reply:
Ok, sounds good.